### PR TITLE
C: Don't write json in read mode

### LINKED ISSUE #151: Serialbox version information in metadata is automatically updated
It seems that serialbox (version 2.3.1) is automatically updating the version number of metadata files (and rendering them unusable in some cases). This is super annoying if the user wants to reuse the files with an older version of serialbox (e.g., 2.3.0). Serialbox asserts when the library version information does not match. 

In my scenario I was using the compare tool of version 2.3.1 that messed up my metadata files.

### LINKED ISSUE #194: compare.py overwrites MetaData JSON files
Assuming the reference serialization data is generated with a certain version (say v2.4.2). Using compare.py from say v2.5.1 will overwrite the MetaData JSON files with a new version.

It is then impossible to reuse the serialization data with the original binary (using version v2.4.2) as reading the serialization data fails due to version mismatch between MetaData and library.

I don't see any reason why a comparison tool should overwrite any of the data.

### PR DESCRIPTION
**Pull request by havogt**
Fixes #194, fixes #151 

Additional changes:
Use the default error handler from python.

### CODE REVIEW

**Comment by havogt** [comment]
launch jenkins

File: src/serialbox-python/serialbox/core.py [hunk 1]
```diff
@@ -75,11 +75,4 @@ def __str__(self):
     def __repr__(self):
         return "<Config {0}>".format(self.instance.compile_options)
 
-def init_serialbox():
```
Comment (clementval) [inline src/serialbox-python/serialbox/core.py]: @havogt We don't need this anymore? It's not clear for me why it's removed...

Comment (havogt) [reply src/serialbox-python/serialbox/core.py]: I have no idea what it did. Apparently we have a 2 ways of writing errors:
- [default is exiting directly on the error](https://github.com/eth-cscs/serialbox2/blob/89768abf2ed08069c1d5aca0ced908c0375bfb7b/src/serialbox-c/ErrorHandling.cpp)
- [the other keeps some error state](https://github.com/eth-cscs/serialbox2/blob/89768abf2ed08069c1d5aca0ced908c0375bfb7b/src/serialbox-c/ErrorHandling.cpp#L62)

The problem with the second one, which was used, is that it segfaults before you get any error. I didn't want to spend time to fix this here.

Comment (havogt) [reply src/serialbox-python/serialbox/core.py]: The result of my change is that we now get an error if something fails from python.

**Pull request merged**